aip/general/0162.md

```diff
@@ -27,94 +27,66 @@ to the version of an API as a whole.
+- Oher resources depend on different revisions of a resource.
+- There is a need to visualize the change of a resource over time.
 
-When it is necessary to refer to a specific revision of a resource, APIs
-**must** use the following syntax: `{resource_name}@{revision_id}`. For
-example:
+### Resource names for revisions
 
-    publishers/123/books/les-miserables@c7cfa2a8
+When referring to specific revision of a resource, APIs must use the following syntax
+`{resource_name}/revisions/{revision_id}`. For example:
 
-**Note:** The `@` character is selected because it is the only character
-permitted by [RFC 1738 §2.2][] for special meaning within a URI scheme that is
-not already used elsewhere.
+publishers/123/books/les-miserables/revisions/c7cfa2a8
 
-APIs **should** generally accept a resource reference at a particular revision
-in any place where they ordinarily accept the resource name. However, they
-**must not** accept a revision in situations that mutate the resource, and
-should error with `INVALID_ARGUMENT` if one is given.
+### Getting a revision
 
-**Important:** APIs **must not** require a revision ID, and **must** default to
-the current revision if one is not provided, except in methods specifically
-dealing with the revision history (such as rollback) where failing to require
-it would not make sense (or lead to dangerous mistakes).
+Services `should` implement the standard `Get` method ([AIP-131][])
 
-### Getting a revision
+### Tagging revisions
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+There are two types of tags: server defined and user specified.
 
-```proto
-message GetBookRequest {
-  // The name of the book.
-  //   Example: publishers/123/books/les-miserables
-  //
-  // In order to retrieve a previous revision of the book, also provide
-  // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
-  string name = 1 [
-    (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
-    }];
-}
-```
+#### Server defined
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+Services **may** reserve specific tags to be server-defined (e.g. `current`, `latest`), which are read-only and managed by the service.
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+- `latest` represents the most recent created revision. The content of `publishers/{publisher}/books/{book}/revisions/latest` and `publishers/{publisher}/books/{book}` can differ. Because the latest revision may be different from the current state of the resource.
```

Comment:
```suggestion
- If a `latest` tag exists, it **must** represents the most recent created revision. The content of `publishers/{publisher}/books/{book}/revisions/latest` and `publishers/{publisher}/books/{book}` can differ. Because the latest revision may be different from the current state of the resource.
```

Fix for AIP styling. also watch line length (80 chars).